src/cluster/slot_migrate.h

```diff
@@ -190,9 +191,9 @@ class SlotMigrator : public redis::Database {
   UniqueFD dst_fd_;
 
   MigrationType migration_type_ = MigrationType::kRedisCommand;
-  std::atomic<int16_t> forbidden_slot_ = -1;
-  std::atomic<int16_t> migrating_slot_ = -1;
-  int16_t migrate_failed_slot_ = -1;
+  std::atomic<SlotRange> forbidden_slot_range_ = SlotRange{-1, -1};
+  std::atomic<SlotRange> slot_range_ = SlotRange{-1, -1};
```

Comment:
should we add a `static_assert(slot_range_.is_lock_free())` to force `slot_range` cas not uses mutex?

Reply:
It seems like this can't be checked at compile time? `is_always_lock_free`?
https://godbolt.org/z/8qeW4M6he

Reply:
Yeah, I think a compile time `is_always_lock_free` is better here, currently two integer will not meet any problem but If future somebody change this to more than 8B this should be take care of